docs/writing-tests/making-a-testing-plan.md

```diff
@@ -0,0 +1,504 @@
+standard](https://fetch.spec.whatwg.org). Here's the algorithm:
+
+> The static error() method, when invoked, must run these steps:
+>
+> 1. Let *r* be a new [Response](https://fetch.spec.whatwg.org/#response)
+>    object, whose
+>    [response](https://fetch.spec.whatwg.org/#concept-response-response) is a
+>    new [network error](https://fetch.spec.whatwg.org/#concept-network-error).
+> 2. Set *r*'s [headers](https://fetch.spec.whatwg.org/#response-headers) to a
+>    new [Headers](https://fetch.spec.whatwg.org/#headers) object whose
+>    [guard](https://fetch.spec.whatwg.org/#concept-headers-guard) is
+>    "`immutable`".
+> 3. Return *r*.
+
+In order to figure out where to write the test (and whether it's needed at
+all), you can review the contents of the `fetch/` directory in WPT. Here's how
+that looks on a UNIX-like command line:
+
+    $ ls fetch
+    api/                           data-urls/   range/
+    content-encoding/              http-cache/  README.md
+    content-length/                images/      redirect-navigate/
+    content-type/                  metadata/    security/
+    corb/                          META.yml     stale-while-revalidate/
+    cors-rfc1918/                  nosniff/
+    cross-origin-resource-policy/  origin/
+
+This test is for a behavior directly exposed through the API, so we should look
+in the `api/` directory:
+
+    $ ls fetch/api
+    abort/  cors/         headers/           policies/  request/    response/
+    basic/  credentials/  idlharness.any.js  redirect/  resources/
+
+And since this is a static method on the `Response` constructor, we would
+expect the test to belong in the `response/` directory:
+
+    $ ls fetch/api/response
+    multi-globals/                   response-static-error.html
+    response-cancel-stream.html      response-static-redirect.html
+    response-clone.html              response-stream-disturbed-1.html
+    response-consume-empty.html      response-stream-disturbed-2.html
+    response-consume.html            response-stream-disturbed-3.html
+    response-consume-stream.html     response-stream-disturbed-4.html
+    response-error-from-stream.html  response-stream-disturbed-5.html
+    response-error.html              response-stream-disturbed-6.html
+    response-from-stream.any.js      response-stream-with-broken-then.any.js
+    response-init-001.html           response-trailer.html
+    response-init-002.html
+
+There seems to be a test file for the `error` method:
+`response-static-error.html`. We can open that to decide if the behavior is
+already covered. If not, then we know where to [write the
+test](https://github.com/web-platform-tests/wpt/pull/19601)!
+
+### Failures on wpt.fyi
+
+There are many behaviors that are difficult to describe in a succinct file
+name. That's commonly the case with low-level rendering details of CSS
+specifications. Test authors may resort to generic number-based naming schemes
```

Comment:
At the time it wasn't a last resort but by convention. It may be best to avoid this discussion here, but @frivoal might have advice on when to use this convention for new tests today.

Reply:
I mean that this section now implicitly discourages numbered file names, and to the best of my understanding it's still a good idea in many cases today. However, if @frivoal doesn't disapprove of the framing here then I could accept it too.

Reply:
It may amuse someone to know that only a few hours later I ran into a problem with numbered tests I hadn't considered: https://github.com/web-platform-tests/wpt/pull/8576#issuecomment-545108563